### ISSUE TITLE
Dynamic max pool for CNN with dynamic sequence length input

### ISSUE THREAD

**Issue by madenindya** [onnx/onnx#3452]
# Ask a Question

### Question
How to do dynamic max_pool in ONNX for CNN with dynamic input sequence_length?

I encounter the wrong shape/size after passing through max_pool when running my onnx model. This happened because the size of my input sample when generating onnx file is different from the size when running the onnx model. (Dynamic seq_len) 

### Further information

This is my model and its `forward` method:
```python

def __init__(...):
    super(MyCNN, self).__init__()

    ...
    # model-cnn with multi window + relu + max pool
    self.word_convs = nn.ModuleList([
        nn.Sequential(
            nn.Conv1d(
                in_channels=embedding_size,
                out_channels=100,
                kernel_size=w,
                padding=w-1,
            ),
            nn.ReLU(inplace=True),
            nn.AdaptiveMaxPool1d(1)  # <<< define max_pool here
        ) for w in kernels_size      # kernel size is dynamic. Example: [1, 2, 3, 5]
    ])
    ...

def forward(self, inp_seq):
    x = self.word_embedding(inp_seq) # bs, seqlen, embeddim
    x = x.transpose(1, 2)            # bs, embeddim, seqlen
    xs = []
    for conv in self.word_convs:
        out = conv(x)           
        out = out.squeeze(2)         # bs, out_channels
        xs.append(out)
    xl = torch.cat(xs, 1)            # bs, out_channels * len(kernels_size)
    xl = self.drop(xl)
    return xl

```

The code I use to convert into onnx is the following:
```python

input_sample = torch.ones([1, 512], dtype=torch.long)
torch.onnx.export(
    model,
    input_sample,
    "model-cnn.onnx",
    export_params=True,
    input_names = ['inp_seq'],   
    output_names = ['output'], 
    dynamic_axes={
        'inp_seq' : {0 : 'batch_size', 1: 'seq_len'},   
        'output' : {0 : 'batch_size', 1: 'seq_len'}},
    verbose=True,
)

```

And it **successfully** produced file `model-cnn.onnx`!

When I run the ONNX model:
- If my **input seq_len is 512** (same as my input_sample) it would **run with no issue.** 
- However, when I try to **run with different seq_len it encounters the RunTime Error** bellow.

```
RuntimeException: [ONNXRuntimeError] : 6 : 
RUNTIME_EXCEPTION : Non-zero status code returned while running Squeeze node. Name:'Squeeze_17' 
Status Message: /onnxruntime_src/onnxruntime/core/providers/cpu/tensor/squeeze.h:49 static std::vector<long int> onnxruntime::SqueezeBase::ComputeOutputShape(const onnxruntime::TensorShape&, const std::vector<long int>&) input_shape[i] == 1 was false. 
Dimension of input 2 must be 1 instead of 0. shape={3,100,0}
```
(3 is my input batch_size, 100 is out_channels from layer conv, 0 should be 1 (from max_pool))

When I try to debug it, I found the **ONNX automatically defines the max_pool even though (I hope) it would be dynamic**. Thus, when the seq_len is different (from the sample input), it would produce a different size after passing the Conv1d layer and made the input for max_pool mismatch. 

```
onnx::MaxPool[kernel_shape=[512], pads=[0, 0], strides=[512]](%15)
onnx::MaxPool[kernel_shape=[513], pads=[0, 0], strides=[513]](%19)
onnx::MaxPool[kernel_shape=[514], pads=[0, 0], strides=[514]](%23)
onnx::MaxPool[kernel_shape=[516], pads=[0, 0], strides=[516]](%27)
```

**What should I do?**

Currently, I handle it by padding the onnx_input to seq_len 512, but do you have any better solution for this problem?
